bin/ruby-build

```diff
@@ -1081,6 +1081,16 @@ has_broken_mac_openssl() {
   [[ "$RUBY_CONFIGURE_OPTS" != *--with-openssl-dir=* ]]
 }
 
+has_openssl_3() {
+  local openssl_version="$(openssl version 2>/dev/null || true)"
```

Comment:
I think we need a better check here because there might be openssl headers yet no openssl executable (https://github.com/rbenv/ruby-build/pull/1974/files#r885043692).

One idea is looking in /usr/include/openssl/opensslv.h, another is to compile a small C program to print `OPENSSL_VERSION_TEXT`.